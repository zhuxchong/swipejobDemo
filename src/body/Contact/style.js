import styled from "styled-components";
export const ContactWrapper = styled.div`
  border-bottom: solid gray 1px;
  width: 100%;
  padding: 3% 0%;
  display: flex;
  flex-direction: column;
  flex: 2;
  h4 {
    margin-top: 0;
    margin-bottom: 10px;
    text-align: center;
    line-height: 20px;
  }

  p {
    margin: 0;
    line-height: 20px;
    text-align: center;
    color: #0066cc;
    text-decoration: underline;
  }
  /*for iphone 5*/
  @media (max-width: 375px) {
    padding: 2% 0%;
    h4 {
      font-size: 15px;
      line-height: 12px;
    }
    p {
      font-size: 10px;
      line-height: 12px;
    }
  }
`;
