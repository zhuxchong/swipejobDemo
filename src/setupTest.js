import { configure } from "enzyme";
import Ada from "enzyme-adapter-react-16";
//set up
configure({ adapter: new Ada() });
