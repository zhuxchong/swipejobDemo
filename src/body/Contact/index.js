import React from "react";
import { ContactWrapper } from "./style";
const Contact = props => {
  const { branch, branchPhoneNumber } = props;
  return (
    <ContactWrapper>
      <h4>Branch</h4>
      <h4 className="branch">{branch}</h4>
      <p className="branchPhoneNumber">{`(${branchPhoneNumber}) 922-4240`} </p>
    </ContactWrapper>
    //<div>123{console.log(props)}</div>
  );
};
export default Contact;
