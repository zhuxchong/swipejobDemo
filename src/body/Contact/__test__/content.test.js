import React from "react";
import { shallow } from "enzyme";
import Contact from "../index";
require("../../../setupTest");
const props = {
  //   title: "Construction and Releated Workers - General",
  //   companyName: "C.D. Barnes and Associates",
  //   wagePerHour: "$8.38/hour",
  //   workPeriod: "Mon, Sep 05 - Fri, Sep 16",
  branch: "Tacoma",
  branchPhoneNumber: "(253) 922-4240"
};
let wrapper;
describe("Description component", () => {
  beforeEach(() => {
    wrapper = shallow(<Contact {...props} />);
  });
  it("component should render correct value", () => {
    expect(wrapper.find(".branch").text()).toEqual(props.branch);
  });

  it("should render the component to match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
