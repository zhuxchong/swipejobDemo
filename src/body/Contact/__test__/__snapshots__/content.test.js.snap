// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description component should render the component to match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Contact
    branch="Tacoma"
    branchPhoneNumber="(253) 922-4240"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h4>
          Branch
        </h4>,
        <h4
          className="branch"
        >
          Tacoma
        </h4>,
        <p
          className="branchPhoneNumber"
        >
          ((253) 922-4240) 922-4240
           
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Branch",
        },
        "ref": null,
        "rendered": "Branch",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Tacoma",
          "className": "branch",
        },
        "ref": null,
        "rendered": "Tacoma",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "((253) 922-4240) 922-4240",
            " ",
          ],
          "className": "branchPhoneNumber",
        },
        "ref": null,
        "rendered": Array [
          "((253) 922-4240) 922-4240",
          " ",
        ],
        "type": "p",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  border-bottom: solid gray 1px;
  width: 100%;
  padding: 3% 0%;
  display: flex;
  flex-direction: column;
  flex: 2;
  h4 {
    margin-top: 0;
    margin-bottom: 10px;
    text-align: center;
    line-height: 20px;
  }

  p {
    margin: 0;
    line-height: 20px;
    text-align: center;
    color: #0066cc;
    text-decoration: underline;
  }
  /*for iphone 5*/
  @media (max-width: 375px) {
    padding: 2% 0%;
    h4 {
      font-size: 15px;
      line-height: 12px;
    }
    p {
      font-size: 10px;
      line-height: 12px;
    }
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h4>
            Branch
          </h4>,
          <h4
            className="branch"
          >
            Tacoma
          </h4>,
          <p
            className="branchPhoneNumber"
          >
            ((253) 922-4240) 922-4240
             
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Branch",
          },
          "ref": null,
          "rendered": "Branch",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Tacoma",
            "className": "branch",
          },
          "ref": null,
          "rendered": "Tacoma",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "((253) 922-4240) 922-4240",
              " ",
            ],
            "className": "branchPhoneNumber",
          },
          "ref": null,
          "rendered": Array [
            "((253) 922-4240) 922-4240",
            " ",
          ],
          "type": "p",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  border-bottom: solid gray 1px;
  width: 100%;
  padding: 3% 0%;
  display: flex;
  flex-direction: column;
  flex: 2;
  h4 {
    margin-top: 0;
    margin-bottom: 10px;
    text-align: center;
    line-height: 20px;
  }

  p {
    margin: 0;
    line-height: 20px;
    text-align: center;
    color: #0066cc;
    text-decoration: underline;
  }
  /*for iphone 5*/
  @media (max-width: 375px) {
    padding: 2% 0%;
    h4 {
      font-size: 15px;
      line-height: 12px;
    }
    p {
      font-size: 10px;
      line-height: 12px;
    }
  }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
